tests/unit/Collectors/JobCollectorTest.php

```diff
@@ -72,7 +72,13 @@ public function testJobProcessingListener()
         $this->agentMock
             ->shouldReceive('startTransaction')
             ->once()
-            ->with(self::JOB_NAME, [], self::REQUEST_START_TIME)
+            ->withArgs(function ($job_name, $context, $start_time) {
+                $this->assertEquals(self::JOB_NAME, $job_name);
+                $this->assertEquals([], $context);
+                $this->assertNotNull($start_time);
```

Comment:
interesting testing problem... would have to spoof `php_sapi_name` or abstract it that it uses the correct time for a specific env.

Reply:
Do you like this? https://marcelog.github.io/articles/php_mock_global_functions_for_unit_tests_with_phpunit.html

Reply:
FWIW I had the same thoughts when using the method and when I looked around the general recommendation was to wrap it in your own method and call/mock the wrapper instead.